consensus/misc/rewards.go

```diff
@@ -3,42 +3,100 @@ package misc
 import (
 	"math/big"
 
+	"github.com/dominant-strategies/go-quai/common"
+	"github.com/dominant-strategies/go-quai/consensus"
 	"github.com/dominant-strategies/go-quai/core/types"
+	"github.com/dominant-strategies/go-quai/params"
+	"modernc.org/mathutil"
 )
 
-func CalculateReward(header *types.WorkObjectHeader) *big.Int {
+func CalculateReward(parent *types.WorkObject, header *types.WorkObjectHeader) *big.Int {
 	if header.PrimaryCoinbase().IsInQiLedgerScope() {
-		return CalculateQiReward(header)
+		return CalculateQiReward(parent.WorkObjectHeader())
 	} else {
-		return CalculateQuaiReward(header)
+		return CalculateQuaiReward(parent)
 	}
 }
 
 // Calculate the amount of Quai that Qi can be converted to. Expect the current Header and the Qi amount in "qits", returns the quai amount in "its"
-func QiToQuai(currentHeader *types.WorkObjectHeader, qiAmt *big.Int) *big.Int {
-	quaiPerQi := new(big.Int).Div(CalculateQuaiReward(currentHeader), CalculateQiReward(currentHeader))
-	return new(big.Int).Mul(qiAmt, quaiPerQi)
+func QiToQuai(parent *types.WorkObject, qiAmt *big.Int) *big.Int {
+	quaiReward := CalculateQuaiReward(parent)
+	quaiByQi := new(big.Int).Mul(quaiReward, qiAmt)
+	qiReward := CalculateQiReward(parent.WorkObjectHeader())
+	return new(big.Int).Quo(quaiByQi, qiReward)
 }
 
 // Calculate the amount of Qi that Quai can be converted to. Expect the current Header and the Quai amount in "its", returns the Qi amount in "qits"
-func QuaiToQi(currentHeader *types.WorkObjectHeader, quaiAmt *big.Int) *big.Int {
-	qiPerQuai := new(big.Int).Div(CalculateQiReward(currentHeader), CalculateQuaiReward(currentHeader))
-	return new(big.Int).Mul(quaiAmt, qiPerQuai)
+func QuaiToQi(header *types.WorkObject, quaiAmt *big.Int) *big.Int {
+	qiByQuai := new(big.Int).Mul(CalculateQiReward(header.WorkObjectHeader()), quaiAmt)
```

Comment:
(nit) QuaiToQi and QiToQuai are not exactly the same line by line, ideally they would be

Reply:
made it look the same